bob/Bob.cs

```diff
@@ -18,86 +18,76 @@ public static string Hey(string statement)
+
         if (IsAllUpperCaseLettersOnly(statement) == true)
             return response[1].ToString();
 
         if (IsAllUpperCaseLettersAndSymbols(statement) == true)
             return response[1].ToString();
 
+        if (ContainsANumberBeforeAWordAndExclamationPoint(statement) == true)
+            return response[1].ToString();
+
         if (ContainsNumberAndQuestionMark(statement) == true)
             return response[2].ToString();
 
         if (IsLowerCaseQuestion(statement) == true)
             return response[2].ToString();
 
-        //if (HasDoubleWhiteSpaceAfterWord(statement) == true)
-        //    return response[2].ToString();
-
-        //if (IsStringEmpty(statement) == true)
-        //    return response[3].ToString();
+        if (ContainsThreeOrMoreSentences(statement) == true)
+            return response[2].ToString();
 
-        //if (IsFourWhiteSpaces(statement) == true)
-        //    return response[2].ToString();
+        if (IsStringNullEmptyOrAllWhiteSpace(statement) == true)
+            return response[3].ToString();
 
 
         return response[0].ToString();
     }
 
+    private static bool ContainsNewLine(string input) => Regex.IsMatch(input, @"\n");
 
+    private static bool IsQuestionMarkNotAtEndOfSentence(string input) => Regex.IsMatch(input, @"\s\?\s");
 
-    static bool IsAllUpperCaseLettersOnly(string input)
-    {
-        var pattern = new Regex(@"^([A-Z]+\s*)+$");
-        var match = pattern.Match(input);
+    private static bool IsAllUpperCaseLettersOnly(string input) => Regex.IsMatch(input, @"^([A-Z]+\s*)+$");
 
-        if (match.Success)
-            return true;
+    private static bool IsAllUpperCaseLettersAndSymbols(string input) => Regex.IsMatch(input, @"^([A-Z]+\s*)+\W$");
 
-        return false;
+    private static bool ContainsANumberBeforeAWordAndExclamationPoint(string input) => Regex.IsMatch(input, @"([0-9])+\s*\w*\!");
 
-    }
+    private static bool ContainsNumberAndQuestionMark(string input) => Regex.IsMatch(input, @"\d\?");
 
-    static bool IsAllUpperCaseLettersAndSymbols(string input)
-    {
-        var pattern = new Regex(@"^([A-Z]+\s*)+\W$");
-        var match = pattern.Match(input);
+    private static bool IsLowerCaseQuestion(string input) => Regex.IsMatch(input, @"[a-z]\w\?");
 
-        if (match.Success)
-            return true;
+    private static bool HasDoubleWhiteSpaceAfterWord(string input) => Regex.IsMatch(input, @"\w\s\s");
 
-        return false;
-    }
-
-    private static bool HasDoubleWhiteSpaceAfterWord(string input)
+    private static bool ContainsThreeOrMoreSentences(string input)
     {
-        var pattern = new Regex(@"\w\s\s");
-        var match = pattern.Match(input);
+        string[] sentences = Regex.Split(input, @"[\.\!\?]");
+        int count = 0;
+
+        foreach (string sentence in sentences)
+        {
+            count++;
+        }
 
-        if (match.Success)
+        if (count >= 3)
```

Comment:
Remember we talked about this. Any time you see this pattern of returning `true/false` based on the outcome of an `if()`, it is redundant